mlir/lib/Dialect/Linalg/Utils/Utils.cpp

```diff
@@ -596,8 +596,17 @@ computeSliceParameters(OpBuilder &builder, Location loc, Value valueToTile,
     auto m = map.getSubMap({r});
     LLVM_DEBUG(llvm::dbgs() << "computeSliceParameters: submap: " << m << "\n");
     IRRewriter rewriter(builder);
-    OpFoldResult offset = makeComposedFoldedAffineApply(rewriter, loc, m, lbs);
+    // The offset of the slice is map(lbs) - map(0).
+    SmallVector<Attribute> zeros(lbs.size(), rewriter.getIndexAttr(0));
+    SmallVector<Attribute> mAtZero;
+    auto res = m.constantFold(zeros, mAtZero);
+    assert(succeeded(res));
```

Comment:
the `res` is not used anyway... So in non-debug builds this will show up as warning/error (if warnings are converted to error). What is the `constantFold` for?

Could you also add a message to the assert.

Reply:
The `m.constantFold` is evaluating the affine map `m` with all inputs being zero. 
The `constantFold` returns `failure()` when the affine map has symbols, but afaik linalg guarantees that the affine map has no symbols, the the evaluation of `m` into a constant is always possible. (Thus the assert; will add a message to it).